### ISSUE TITLE
Assigning `{}` doesn't zero fields correctly without optimizations

### ISSUE THREAD

**Issue by Tetralux** [odin-lang/Odin#790]
* Operating System: Linux x64
* Please paste `odin version` output: `odin version 0.13.1` (commit ca4b0527)

The example from https://github.com/odin-lang/Odin/issues/789, when compiled _without optimizations_, doesn't correctly zero the fields:

```odin
package test

import "core:mem"
import "core:fmt"

main :: proc() {
    using fmt;

    S :: struct {
        i: i32,
        u: u64,
    };

    pb :: proc(s: ^S) {
        data := mem.ptr_to_bytes(s, 1);
        fmt.printf("%x\n", data);
    }

    s: S;
    pb(&s); println(s); println();

    data := mem.ptr_to_bytes(&s, 1);
    for b in &data {
        b = 0xaa;
    }
    pb(&s); println(s); println();

    s = {};
    pb(&s); println(s); println();
    /*
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, aa, aa, aa, aa]
    S{i = 0, u = 12297829379609722880}
    */
}
```

**Comment by gingerBill** [comment]
Is this still happening?

**Comment by Lord-Remington-the-3rd** [comment]
Yes, I have the same compiler version and OS.
Interstingly, even in release the bytes seem to print out non zero? Tho the struct and members are zero when tested (print(s.u) == 0):
[0, 0, 0, 0, aa, aa, aa, aa, 0, 0, 0, 0, 0, 0, 0, 0]
S{i = 0, u = 0}

**Comment by Lord-Remington-the-3rd** [comment]
I also just updated to the latest nightly. The problem still persists. @gingerBill

**Comment by gingerBill** [comment]
Goddamn it. I know what I'll have to do to fix it.

**Comment by gingerBill** [comment]
Could you please test this on the latest commit.

**Comment by Tetralux** [comment]
Fixed as of d7a5767aa3497f9b4b6f0ced955ce31fb019ef80.
